static/js/search-traces.js

```diff
@@ -16,167 +16,138 @@
+    if (urlParams.has('minDuration')) state.minDuration = urlParams.get('minDuration');
+    if (urlParams.has('maxDuration')) state.maxDuration = urlParams.get('maxDuration');
+    if (urlParams.has('limit')) state.limit = urlParams.get('limit');
+    if (urlParams.has('tags')) state.tags = urlParams.get('tags');
+
+    // Update UI with state values
+    $('#service-span-name').text(state.service);
+    $('#operation-span-name').text(state.operation);
+    $('#min-duration-input').val(state.minDuration);
+    $('#max-duration-input').val(state.maxDuration);
+    $('#limit-result-input').val(state.limit);
+    $('#tags-input').val(state.tags);
+    const start = state.startEpoch;
+    const end = state.endEpoch;
+    $(`.ranges .inner-range .range-item`).removeClass('active');
+    if (!isNaN(start)) {
+        let stDate = Number(start);
+        let endDate = Number(end);
+        datePickerHandler(stDate, endDate, 'custom');
+        loadCustomDateTimeFromEpoch(stDate, endDate);
     } else {
-        const urlParams = new URLSearchParams(window.location.search);
-        if (urlParams.toString() && !initialSearchPerformed) {
-            setTimeout(() => {
-                searchTraceHandler(new Event('init'));
-            }, 300);
-        }
+        $(`.ranges .inner-range #${start}`).addClass('active');
+        datePickerHandler(start, end, start);
     }
-};
-
-function initPage() {
-    initChart(true);
-    // Initialize dropdowns with URL values first
-    initializeDropdowns();
-    handleSort();
-    handleDownload();
-    handleTimePicker();
-    $('#search-trace-btn').on('click', searchTraceHandler);
+    return state;
 }
 
-function initializeDropdowns() {
-    if (dropdownsInitialized) {
-        return;
-    }
-    const state = getInitialSearchState();
-    const serviceValue = state.service || 'All';
-    const operationValue = state.operation || 'All';
-
-    dropdownsInitialized = true;
-
-    getValuesOfColumn('service', 'Service', serviceValue);
+function initDropdowns() {
+    $('#service-dropdown').singleBox({
+        spanName: 'Service',
+        dataList: ['All'],
+        defaultValue: $('#service-span-name').text() || 'All',
+        dataUpdate: true,
+        clickedHead: async function () {
+            return await fetchColumnValues('service');
+        },
+    });
 
-    setTimeout(() => {
-        getValuesOfColumn('name', 'Operation', operationValue);
-    }, 100);
+    $('#name-dropdown').singleBox({
+        spanName: 'Operation',
+        dataList: ['All'],
+        defaultValue: $('#operation-span-name').text() || 'All',
+        dataUpdate: true,
+        clickedHead: async function () {
+            return await fetchColumnValues('name', $('#service-span-name').text());
+        },
+    });
 }
 
+async function fetchColumnValues(column, serviceFilter = null) {
+    let searchText = `SELECT DISTINCT \`${column}\` FROM \`traces\``;
```

Comment:
not in this PR, but in the next one, we should replace these SQL queries with SPL